controller/contour/controller_test.go

```diff
@@ -116,3 +118,12 @@ var _ = Describe("Run controller", func() {
 		})
 	})
 })
+
+func checkContainerHasImage(t *testing.T, container *corev1.Container, image string) {
+	t.Helper()
+
+	if container.Image == image {
```

Comment:
This checks against the image defined in the flag right? So the version of the image defined in the flag changes, would this then cause an immediate upgrade?

Reply:
As mentioned above, the contour/envoy container images are set from the operator CLI. If the operator was restarted with new values for these flags, the images would be differ causing the Envoy `DamonSet` and the Contour `Deployment` to trigger a rolling update.

Reply:
Maybe it's a docs issue then but that behavior needs to be super clear to users. I would have expected that upgrades to the instances would be done via the CRD that manages that instance.

Reply:
+1 to @stevesloka's comment, I think ideally you'd manage the versions of Contour and Envoy on the CRD so they could be controlled declaratively and independently of one another if multiple Contours are being managed. Let's file an issue to track.

Reply:
https://github.com/projectcontour/contour-operator/issues/55